src/components/application_manager/test/application_manager/application_manager_test.cc

```diff
@@ -0,0 +1,138 @@
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+#include <stdint.h>
+#include <set>
+
+#include "gtest/gtest.h"
+
+#include "application_manager/application_manager_impl.h"
+#include "application_manager/mock_application_manager_settings.h"
+#include "application_manager/mock_application.h"
+#include "application_manager/mock_message_helper.h"
+#include "application_manager/mock_resume_ctrl.h"
+#include "application_manager/mock_hmi_capabilities.h"
+#include "application_manager/policies/mock_policy_handler_interface.h"
+#include "media_manager/mock_media_manager.h"
+#include "resumption/last_state.h"
+#include "policy/mock_policy_settings.h"
+#include "utils/shared_ptr.h"
+#include "utils/make_shared.h"
+#include "utils/file_system.h"
+
+namespace am = application_manager;
+namespace policy_test = test::components::policy_handler_test;
+using ::test::components::application_manager_test::MockApplication;
+
+using ::testing::_;
+using ::testing::Return;
+using ::testing::ReturnRef;
+using ::testing::NiceMock;
+
+namespace resumption {
+class LastState;
+}
+
+namespace test {
+namespace components {
+namespace application_manager_test {
+
+namespace {
+const std::string kAppStorageFolder("./storage");
+const std::string kAppInfoStorage("info");
+const uint32_t kThreadPoolSize = 1;
+const uint32_t kAppId = 10u;
+const bool kHmiLaunch = false;
+const uint32_t kStopStreamingTimeout = 10000;
+const std::string kFileName = "file_name";
+}  // namespace
+
+class ApplicationManagerImplTest : public ::testing::Test {
+ public:
+  void SetUp() OVERRIDE {
+    EXPECT_CALL(mock_application_manager_settings_, thread_pool_size())
+        .WillOnce(Return(kThreadPoolSize));
+    EXPECT_CALL(mock_application_manager_settings_,
+                hmi_capabilities_file_name()).WillOnce(ReturnRef(kFileName));
+    EXPECT_CALL(mock_application_manager_settings_, use_db_for_resumption())
+        .WillRepeatedly(Return(false));
+    EXPECT_CALL(mock_application_manager_settings_, launch_hmi())
+        .WillOnce(Return(kHmiLaunch));
+    EXPECT_CALL(mock_application_manager_settings_, stop_streaming_timeout())
+        .WillRepeatedly(Return(kStopStreamingTimeout));
+    EXPECT_CALL(mock_application_manager_settings_,
+                app_resumption_save_persistent_data_timeout())
+        .WillOnce(ReturnRef(kAppId));
+    EXPECT_CALL(mock_application_manager_settings_, max_number_of_ios_device())
+        .WillRepeatedly(Return(0));
+    EXPECT_CALL(mock_policy_settings_, enable_policy())
+        .WillOnce(Return(true))
+        .WillOnce(Return(false));
```

Comment:
@okozlovlux I haven't seen such duplication of WillOnce() statement, is this expectation correct?